nipype/interfaces/freesurfer/longitudinal.py

```diff
@@ -251,3 +253,207 @@ def _list_outputs(self):
+
+
+    @property
+    def cmdline(self):
+        cmd = super(BaseReconAll, self).cmdline
+
+        if not self._is_resuming():
+            return cmd
+
+        subjects_dir = self.inputs.subjects_dir
+        if not isdefined(subjects_dir):
+            subjects_dir = self._gen_subjects_dir()
+
+        directive = self.inputs.directive
+        if not isdefined(directive):
+            steps = []
+
+        iflogger.info(f"recon-all: {cmd}")
+        return cmd
+
+
+class LongReconAllInputSpec(ReconAllInputSpec):
+    subject_id = traits.Str(argstr="-subjid %s", desc="subject name")
+    long_id = traits.Str(
+        argstr="-long %s",
+        desc="longitudinal session/timepoint id",
+        xor=["subject_id"],
+        requires=["base_id"],
+        position=1
+    )
+    base_id = traits.Str(
+        argstr="%s",
+        desc="longitudinal base template id",
+        requires=["long_id"],
+        position=2
+    )
+
+
+class LongReconAllOutputSpec(FreeSurferSource.output_spec):
+    subjects_dir = Directory(exists=True, desc="FreeSurfer subjects directory")
+    subject_id = traits.Str(desc="Subject template name")
+
+
+class LongReconAll(CommandLine):
+    """Uses FreeSurfer's longitudinal recon-all to process a subject given
+    the previously processed base template.
+
+    Examples
+    ---------
+
+    >>> from nipype.interfaces.freesurfer.longitudinal import LongReconAll
+    >>> longrecon = LongReconAll()
+    >>> longrecon.inputs.long_id = "ses-1"
+    >>> longrecon.inputs.base_id = "sub-template"
+    >>> longrecon.inputs.directive = "all"
+    >>> longrecon.inputs.subjects_dir = "."
+    >>> longrecon.cmdline
+    'recon-all -all -long ses-1 sub-template -sd .'
+    """
+
+    _cmd = "recon-all"
+    input_spec = LongReconAllInputSpec
+    output_spec = LongReconAllOutputSpec
+    _can_resume = True
+    force_run = False
+
+    def _gen_subjects_dir(self):
+        return os.getcwd()
+
+    def _gen_filename(self, name):
+        if name == "subjects_dir":
+            return self._gen_subjects_dir()
+        return None
+
+    def _list_outputs(self):
+        subject_id = f"{self.inputs.long_id}.long.{self.inputs.base_id}"
+
+        if isdefined(self.inputs.subjects_dir):
+            subjects_dir = self.inputs.subjects_dir
+        else:
+            subjects_dir = self._gen_subjects_dir()
+
+        if isdefined(self.inputs.hemi):
+            hemi = self.inputs.hemi
+        else:
+            hemi = "both"
+
+        outputs = self._outputs().get()
+
+        outputs.update(
+            FreeSurferSource(
+                subject_id=subject_id, subjects_dir=subjects_dir, hemi=hemi
+            )._list_outputs()
+        )
+        outputs["subject_id"] = subject_id
+        outputs["subjects_dir"] = subjects_dir
+        return outputs
+
+    def _is_resuming(self):
+        subjects_dir = self.inputs.subjects_dir
+        subject_id = f"{self.inputs.long_id}.long{self.inputs.base_id}"
```

Comment:
Missing dot in computed `subject_id`:

```suggestion
        subject_id = f"{self.inputs.long_id}.long.{self.inputs.base_id}"
```